src/tiled/commanddatamodel.cpp

```diff
@@ -451,8 +453,10 @@ QKeySequence CommandDataModel::shortcut(const QModelIndex &index) const
 
 void CommandDataModel::setShortcut(const QModelIndex &index, const QKeySequence &keySequence)
 {
-    if (index.row() < mCommands.size())
+    if (index.row() < mCommands.size()) {
         mCommands[index.row()].shortcut = keySequence;
+        emit dataChanged(index, index);
```

Comment:
Actually you can't necessarily re-use the given index here, since you need to make sure to tell the view that the `ShortcutColumn` changed. If you test it, you'll notice that it only works if you selected the command by clicking on the shortcut column.

So you need to do:
```c++
QModelIndex shortcutIndex = this->index(index.row(), ShortcutColumn);
emit dataChanged(shortcutIndex, shortcutIndex);
```